process_css_extend_template.xml

```diff
@@ -4,38 +4,34 @@
 --><project xmlns:dita="http://dita-ot.sourceforge.net" name="fox.jason.extend.css">
     
     <target name="extend.css.init">
+        <condition property="user.csspath" value="" else="${args.csspath}/">
+            <or>
+                <not>
+                    <isset property="args.csspath"/>
+                </not>
+                <isabsolute path="${args.csspath}"/>
+            </or>
+        </condition>
         <mkdir dir="${dita.temp.dir}"/>
-        <tempfile deleteonexit="true" destdir="${dita.temp.dir}" property="extend.css.file" suffix=".css"/>
-        <touch file="${extend.css.file}"/>
+        <tempfile deleteonexit="true" destdir="${dita.temp.dir}" property="extend.css.file" suffix=".css" createfile="yes"/>
     </target>
     
     <target name="extend.css.copy" >
         <copy tofile="${output.dir}/${user.csspath}/common-extended.css" file="${extend.css.file}"/>
     </target>
-    
-    <target name="extend.css" depends="extend-css-check-transtype, extend-css-set-properties" if="enable.extend.css">
-        <antcall target="call.extend.css" inheritRefs="true"/>
-    </target>
-    
+
     <target name="extend-css-check-transtype">
         <condition property="enable.extend.css">
             <or>
+                <contains string="${out.ext}" substring="htm" casesensitive="no"/>
                 <contains string="${transtype}" substring="HTML" casesensitive="no"/>
                 <contains string="${transtype}" substring="eclipsehelp" casesensitive="no"/>
             </or>
         </condition>
     </target>
-    
-    <target name="extend-css-set-properties" depends="extend-css-check-transtype" if="enable.extend.css">
-        <condition property="user.csspath" value="">
-            <or>
-                <not>
-                    <isset property="args.csspath"/>
-                </not>
-                <isabsolute path="${args.csspath}"/>
-            </or>
-        </condition>
-        <property name="user.csspath" value="${args.csspath}/"/>
+
+    <target name="extend.css" depends="extend-css-check-transtype" if="enable.extend.css">
```

Comment:
there's no reason to do all the computing of css if you don't copy it at the end, the whole extend.css chain should be skipped

Reply:
Calculation shouldn't be in this plugin anyway - removed